### ISSUE TITLE
miller_index_from_sites gives wrong miller index

### ISSUE THREAD

**Issue by utf** [materialsproject/pymatgen#1301]
The `pymatgen.core.surface.miller_index_from_sites` method gives incorrect results. It also seems quite temperamental about the choice of coordinates. E.g.

```python
from pymatgen.core.structure import Structure
from pymatgen.core.surface import miller_index_from_sites

structure = Structure(
    [10, 0, 0, 0, 10, 0, 0, 0, 10],
    ['Fe', 'Fe', 'Fe'],
    [[0.5, 0.8, 0.8], [0.5, 0.4, 0.2], [0.5, 0.3, 0.7]]
)

miller_index_from_sites(structure.lattice.matrix, structure.cart_coords)
```

Gives `[-1.04,  0.  ,  0.  ]` rather than `[1,  0.  ,  0.  ]`.

Also, changing the atom coordinates slightly to:

```python
[[0.5, 0.8, 0.8], [0.5, 0.6, 0.6], [0.5, 0.3, 0.3]]
```

Gives `[nan, nan, nan]`.

**Comment by richardtran415** [comment]
Looking into it

**Comment by utf** [comment]
I've got a fix for this. Just doing some testing but will submit a PR tomorrow.

**Comment by utf** [comment]
Fixed in #1302
